### PR TITLE
#424 secretd enclave working

### PR DESCRIPTION
**Pull request by Cashmaney**
#424 Added a small health check that prints out the enclave status

### CODE REVIEW

File: cosmwasm/packages/enclave-ffi-types/src/types.rs [hunk 1]
```diff
@@ -165,6 +165,21 @@ pub enum NodeAuthResult {
...
 }
 
+/// This type represents the possible error conditions that can be encountered in the
+/// enclave while authenticating a new node in the network.
+/// cbindgen:prefix-with-name
+#[repr(C)]
+#[derive(Debug, Display, PartialEq, Eq)]
+pub enum HealthCheckResult {
+    Success,
+}
```
Comment (reuvenpo) [inline cosmwasm/packages/enclave-ffi-types/src/types.rs:175]: This enum only has one variant? that's odd

File: cosmwasm/packages/wasmi-runtime/src/exports.rs [hunk 2]
```diff
@@ -181,3 +181,11 @@ pub unsafe extern "C" fn ecall_query(
...
     }
 }
+
+/// # Safety
+/// Always use protection
+#[no_mangle]
+pub unsafe extern "C" fn ecall_health_check(
+) -> HealthCheckResult {
+    HealthCheckResult::Success
+}
```
Comment (reuvenpo) [inline cosmwasm/packages/wasmi-runtime/src/exports.rs:191]: Is this intentionally empty? You could add some checks here, theoretically.
also, why the newline between the function definition's `()`s?

File: cosmwasm/packages/enclave-ffi-types/src/types.rs (hunk 1 above)
Comment (Cashmaney) [reply cosmwasm/packages/enclave-ffi-types/src/types.rs:175]: One day we might want to make this check other things and return more detailed responses. So it's an enum for easy upgradability

File: cosmwasm/packages/wasmi-runtime/src/exports.rs (hunk 2 above)
Comment (Cashmaney) [reply cosmwasm/packages/wasmi-runtime/src/exports.rs:191]: Intentionally empty, yes. When we want to check other things we can add to it, but right now I wanted something that just reports if the enclave is running fine or not

**Pull request merged**